src/components/HideContent/HideContent.tsx

```diff
@@ -0,0 +1,92 @@
+import React, {
+  memo,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+  VFC,
+} from 'react';
+import { Box, HStack } from '@chakra-ui/react';
+import { TextButton } from '../TextButton/TextButton';
+import { assertNonNull } from '../../utils/assertNonNull';
+
+interface Props {
+
+  /**
+   * Max height of visible content in pixels.
+   * When it is exceeded, the show content button appears.
+   */
+  readonly maxHeight: number;
+
+  /** Children prop. */
+  readonly children: ReactNode;
+
+  /** Duration of show/hide transition in milliseconds. */
+  readonly transitionDuration?: number;
+
+  /** Text on the show content button. */
+  readonly showLabel?: string;
+
+  /** Text on the hide content button. */
+  readonly hideLabel?: string;
+}
+
+/**
+ * Component that hides sections of content with long height.
+ */
+const HideContentComponent: VFC<Props> = ({
+  maxHeight,
+  transitionDuration = 300,
+  showLabel = 'Show more',
+  hideLabel = 'Hide',
+  children,
+}) => {
+  const [isHidden, setIsHidden] = useState(true);
+  const [shouldButtonsBeRendered, setShouldButtonsBeRendered] = useState(true);
+  const contentContainerRef = useRef<HTMLDivElement | null>(null);
```

Comment:
I suggest using `typeof Box` instead of HTMLDivElement. Maybe there even better alternative in chakra.

Reply:
`typeof Box` and `HTMLDivElement` are different things entirely